x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/status.tsx

```diff
@@ -0,0 +1,195 @@
+        console.error(err);
+        throw new Error(`Could not instantiate a createdDate object from: ${session.created}`);
+      }
+
+    case STATUS.EXPIRED:
+      try {
+        const expiredOnDate = dateString(session.expires!, timezone);
+        const toolTipContent = i18n.translate(
+          'xpack.data.mgmt.searchSessions.status.message.expiredOn',
+          {
+            defaultMessage: 'Expired on {expiredOnDate}',
+            values: { expiredOnDate },
+          }
+        );
+
+        return {
+          icon: <EuiIcon color="#9AA" type="clock" />,
+          label: <TableText>{getStatusText(session.status)}</TableText>,
+          toolTipContent,
+        };
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+        throw new Error(`Could not instantiate a expiration Date object from: ${session.expires}`);
+      }
+
+    case STATUS.CANCELLED:
+      return {
+        icon: <EuiIcon color="#9AA" type="crossInACircleFilled" />,
+        label: <TableText>{getStatusText(session.status)}</TableText>,
+        toolTipContent: i18n.translate('xpack.data.mgmt.searchSessions.status.message.cancelled', {
+          defaultMessage: 'Cancelled by user',
+        }),
+      };
+
+    case STATUS.ERROR:
+      return {
+        textColor: 'danger',
+        icon: <EuiIcon color="danger" type="crossInACircleFilled" />,
+        label: <TableText>{getStatusText(session.status)}</TableText>,
+        toolTipContent: i18n.translate('xpack.data.mgmt.searchSessions.status.message.error', {
+          defaultMessage: 'Error: {error}',
+          values: { error: (session as any).error || 'unknown' },
+        }),
+      };
+
+    case STATUS.COMPLETE:
+      try {
+        const expiresOnDate = dateString(session.expires!, timezone);
+        const toolTipContent = i18n.translate('xpack.data.mgmt.searchSessions.status.expiresOn', {
+          defaultMessage: 'Expires on {expiresOnDate}',
+          values: { expiresOnDate },
+        });
+
+        return {
+          textColor: 'secondary',
+          icon: <EuiIcon color="secondary" type="checkInCircleFilled" />,
+          label: <TableText>{getStatusText(session.status)}</TableText>,
+          toolTipContent,
+        };
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+        throw new Error(
+          `Could not instantiate a expiration Date object for completed session from: ${session.expires}`
+        );
+      }
+
+      // Error was thrown
+      return null;
+
+    default:
+      throw new Error(`Unknown status: ${session.status}`);
+  }
+};
+
+export const StatusIndicator = (props: StatusIndicatorProps) => {
+  try {
+    const statusDef = getStatusAttributes(props);
+    const { session } = props;
+
+    if (statusDef) {
+      const { toolTipContent } = statusDef;
+      let icon: ReactElement | string | undefined = statusDef.icon;
```

Comment:
Wouldn't it be nicer to create a separate `const` for the `ReactElement`?

Reply:
Not sure I understand